### DISCUSSION TITLE
Error on Plus Server Launcher: "No BIRDs were found anywhere", Failed to connect SonixGPS Type="Ascension3DG" on SonixTablet

### DISCUSSION THREAD

**Discussion by SenonETS** [PlusToolkit/PlusLib#816] [Q&A] [answered]
Hello,

This a newbie on PLUS.

I installed PlusApp-2.8.0.20191105-Win32.exe, and it _works well to connect Type="SonixVideo"_.

This is the instruction I was following [this official tutorial](http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceAscension3DG.html)
I disabled the license of **SonixGPS**, however, always failed to connect to the TrackerDevice, Type="**Ascension3DG**" .

Here is the log:

https://ibb.co/WWCkzvz

060221_092322.008|ERROR|2676.149000|SERVER> System : No BIRDs were found anywhere| in :\d\psnpb\pluslib\src\plusdatacollection\ascension3dgtracking\vtkPlusAscension3DGTrackerBase.cxx(462)
060221_092322.009|ERROR|2676.150000|SERVER> System : InitializeBIRDSystem() not called yet| in :\d\psnpb\pluslib\src\plusdatacollection\ascension3dgtracking\vtkPlusAscension3DGTrackerBase.cxx(462)
060221_092322.009|ERROR|2676.150000|SERVER> Connection initialization failed| in :\d\psnpb\pluslib\src\plusdatacollection\ascension3dgtracking\vtkPlusAscension3DGTrackerBase.cxx(88)
060221_092322.010|ERROR|2676.151000|SERVER> TrackerDevice: Cannot connect to data source, ConnectInternal failed| in :\D\PSNPb\PlusLib\src\PlusDataCollection\vtkPlusDevice.cxx(1147)
060221_092322.010|ERROR|2676.151000|SERVER> Unable to connect device: TrackerDevice.| in :\D\PSNPb\PlusLib\src\PlusDataCollection\vtkPlusDataCollector.cxx(353)
060221_092322.511|ERROR|2676.652000|SERVER> Datacollector failed to connect to devices| in :\D\PSNPb\PlusLib\src\PlusServer\Tools\PlusServer.cxx(106)
060221_092322.611|ERROR|2676.752000| Server stopped unexpectedly. Return code: 1| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(864)
060221_092322.611|INFO|2676.752000| Disconnect request successful| in E:\D\PSNPb\PlusApp\PlusServerLauncher\PlusServerLauncherMainWindow.cxx(632)

Could anyone share some ideas?

**Comment by lassoan** [comment, marked as answer]
Have you followed all the installation instructions described [here](http://perk-software.cs.queensu.ca/plus/doc/nightly/user/DeviceSonixVideo.html)?

**Reply by SenonETS** [reply]
Yes, as I said above, it works with Type="SonixVideo" only;
as long as I add the TrackerDevice, Type="Ascension3DG", I get the error and failed to connect.

The error is always the same, starting with "No BIRDs were found anywhere". 
You can see the screenshot from this [link](https://ibb.co/WWCkzvz), where SL__SonixGPS is the ID of device with Type="Ascension3DG" .

**Reply by lassoan** [reply]
First check connection with the EM tracker only. Shut down the Ultrasonix Exam software (as it can prevent access to the EM tracker if not configured correctly).

Were the ascension3dg DLL files the same in Plus as in the Ultrasonix Exam software?

Can the Ascension cubes demo application connect to the tracker?

**Reply by SenonETS** [reply]
Actually, I have really rare experience with the Ultrasonix machine. Everything was connected before I touch it.

Could you please share any connection instruction link for the connection of the EM tracker? 

Secondly, how exactly should I shut down the Ultrasonix Exam software? The only thing I know is to press "Alt + F4" to shut down the operation system of the SonixTablet. 

Thirdly, how to find the ascension3dg DLL files in the Ultrasounix Exam software? This refers to the SonixTablet running XP 32bit OS, right?
And how to find it in PLUS? Does this "PLUS" refer to the PLUS running Windows10 64bit OS ?

Fourthly, what is the Ascension cubes demo? How does that application connect to the tracker? Does this tracker refer to the AscensionBay (SonixGPS hardware)?

I appreciate your help solving my questions!

**Reply by lassoan** [reply]
From what you describe it seems that you are running Plus on a different computer. If you need live EM tracking information then you must run PlusServer on the SonixTablet.

> Could you please share any connection instruction link for the connection of the EM tracker?

If you have any specific questions let us know, but probably it is better to ask someone in your lab who has more experience. It is hard to explain how and where to plug cables remotely.

You can get Ascension tracker documentation, driver, etc. from [NDI Support website](https://support.ndigital.com/).

> how exactly should I shut down the Ultrasonix Exam software? The only thing I know is to press "Alt + F4" to shut down the operation system of the SonixTablet.

You may need to switch to research mode to prevent shutting down the Exam software from shutting down the entire computer.

> Thirdly, how to find the ascension3dg DLL files in the Ultrasounix Exam software? This refers to the SonixTablet running XP 32bit OS, right?

Search for all the ascension*.dll files that you have within the Plus installation folder on the SonixTablet.

> what is the Ascension cubes demo? How does that application connect to the tracker? Does this tracker refer to the AscensionBay (SonixGPS hardware)?

It is a demo application from NDI/Ascension. It is included in the SDK package that you can download from the NDI support website. It connects to the EM tracker the same way as Plus (running on the SonixTablet). If it cannot connect then it means that there is a driver incompatibility issue (firmware/driver of the EM tracker is not compatible with the SDK that the the cube demo is using).

**Reply by SenonETS** [reply]
Just to make sure that we are on the same page.

My working PC is running Windows10 64bit OS, with PLUS, Slicer installed.
The SonixTablet is running Windows XP 32bit OS, running Ultrasounix SDK 6.0.7, like this [image]( https://cdn.bimedis.com/search/aimage/standard/1298639
)

> If you need live EM tracking information then you must run PlusServer on the SonixTablet.

Do you mean that the configuration file with Type="Ascension3DG" is supposed to be functional **only** on the SonixTablet on Windows XP 32bit OS?

If so, does that mean I will have to run corresponding Slicer Python scripts on the SonixTablet? Even for AI online training related stuff?

**Reply by lassoan** [reply]
The EM tracker is plugged into the SonixTablet. You cannot access that piece of hardware from a different computer. PlusServer must run on the SonixTablet, which has 32-bit Windows XP embedded. This computer is very weak by today's standard and you should not run anything else on it than the Ultrasonix Exam software (if you use the Ulterius interface) and PlusServer. PlusServer sends all data via the network to other computers for processing.

**Reply by SenonETS** [reply]
> This computer is very weak by today's standard and you should not run anything else on it than the Ultrasonix Exam software (if you use the Ulterius interface) and **PlusServer**. **PlusServer** sends all data via the network to other computers for processing.

In the quote, you mentioned **PlusServer** for two times. Are those two **PlusServer** the same?
1. Are they both running on the weak SonixTablet?
2. Are they referring the same configuration file?
3. Are they functioning two jobs at the same time (receiving Ascension3DG tracking data, and sending MixedStream to a strong Windows10 64bit PC through the internet)?

**Reply by ungi** [reply]
No, there is only one PlusServer, running on the Ultrasonix (Windows XP) machine. That PlusServer is reading images from the Sonix software (running in research mode), and reading tracking data from the Ascension tracker. You need another computer on the same network (LAN) to run another application (typically 3D Slicer). 3D Slicer will read the tracked images over the network from PlusServer. 3D Slicer will not run on the Ultrasonix machine, because that machine is very old.

**Reply by SenonETS** [reply]
Thank you ungi! 

>  there is only one PlusServer, running on the Ultrasonix (Windows XP) machine. ... You need another computer on the same network (**LAN**) to run another application (typically 3D Slicer).  **3D Slicer** will read the tracked images over the network from PlusServer.

I am still a little bit confused, please help me confirm:
1. **3D Slicer** will only run on a strong 64bit PC, instead of the SonixTablet, right?
2. To read the tracked images over the network from PlusServer, does the strong 64bit PC connect to the SonixTablet **only** by **LAN**  (both PC and SonixTablet **lose** the internet in this way), or it can also be connected through an IP address through a regional network (e.g. VPN) where both the strong PC and the SonixTablet are connected to the internet?
3. In #2, does it mean using OpenIGTLinkIF module in Slicer to read the tracked images? If so, how to set up the "Hostname" (not localhost any more, right?) and "Port"?

**Reply by lassoan** [reply]
> 3D Slicer will only run on a strong 64bit PC, instead of the SonixTablet, right?

yes

> To read the tracked images over the network from PlusServer, does the strong 64bit PC connect to the SonixTablet only by LAN

You can connect the Slicer computer to SonixTablet using either private or public network. However, connecting a Windows XP computer to the public internet is not a good idea due to known, widely known and exploited vulnerabilities. Your IT department would probably not even let you do it, because it would pose such high security risk for the entire organization.

> does it mean using OpenIGTLinkIF module in Slicer to read the tracked images? If so, how to set up the "Hostname" (not localhost any more, right?) and "Port"?

This is standard TCP/IP networking. You can use any hostname or IP address. If you don't want to configure network settings in your computer and SonixTablet then the simplest is to connect the two computers using a router. That way the router assigns IP addresses, manages DNS. You can also configure your router to allow internet access to the Slicer computer but cut off the SonixTablet from the internet.

**Reply by SenonETS** [reply]
Thank you so much for your help, Iassoan!

It seems I can connect now. However, in Slicer, I got still only "Image_Transducer" node in the Subject hierarchy of module "Data", which is no different than retrieving the Type="SonixVideo" only configuration; in other words, I was not able to find any expected information from the Type="Ascension3DG" configuration.

Could you please help?

**Reply by lassoan** [reply]
The original topic (no birds found) is now resolved. Let's close this issue then please create a new one for get help with receiving transforms. Don't forget to attach your Plus config file.
